providers/base.ts

```diff
@@ -351,6 +351,48 @@ export abstract class ReleaseLookup<Provider extends MetadataProvider, RawReleas
 	}
 }
 
+/** Extends MetadataProvider with functions common to lookups accessing web APIs. */
+export abstract class MetadataApiProvider extends MetadataProvider {
+	/** Must be implemented to perform a request against the specified URL. */
+	abstract query<Data>(apiUrl: URL, maxTimestamp?: number): Promise<CacheEntry<Data>>;
+}
```

Comment:
Good idea to make this an explicit requirement, almost all providers had this anyway.